catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchSource.java

```diff
@@ -936,4 +950,19 @@ private void applyFilters(OpenSearchFilterVisitorObject visitor, WebClient clien
             client.replaceQueryParam(URL_SRC_PARAMETER, "");
         }
     }
+
+    private List<Metacard> processAdditionalForeignMarkups(Element element, String id) {
+        List<Metacard> metacards = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(markUpList)) {
+            XMLOutputter xmlOutputter = new XMLOutputter();
+            for (String markup : markUpList) {
+                if (element.getName().equals(markup)) {
```

Comment:
Looking at the JDOM API, it looks like the element will only have one name http://www.jdom.org/docs/apidocs/org/jdom2/Element.html#getName().  I also don't think this list will ever exceed a handful of element names.